### PR TITLE
Update generate data script to seed mainnet data

### PR DESCRIPTION
**Pull request by njgheorghita**
- Renamed `generate-data.rs` to `seed-database.rs` and moved it to top-level `/src/bin/` since that helps simplify Ansible playbook configuration
- Updated `seed-database.rs` script to iterate through the `mainnetMM` data dump and seed a trin database with content.
- Adds `portal_stateLocalContent` & `portal_historyLocalContent` json rpc endpoints to help with debugging testnets & their local data

### CODE REVIEW

File: trin-core/src/portalnet/storage.rs [hunk 1]
```diff
@@ -133,6 +133,7 @@ impl PortalStorage {
         let distance_to_content_id = self.distance_to_content_id(&content_id);
 
         // Check whether data is outside our radius.
+        // Why is this returning ok?
         if distance_to_content_id > self.data_radius {
             debug!("Not storing: {}", key);
             return Ok(());
```
[the remark this answers is not in the document]
Comment (mrferris) [reply trin-core/src/portalnet/storage.rs]: Good question. I had been thinking of returning `Ok(())` as "I have successfully done my job of storing (or not storing) data that I should store (or should not store!)". I agree that that's confusing/brittle, and we should return an error in this case.

File: src/bin/seed-database.rs [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+
+/// Type for data format stored in mainnetMM data dump
+#[derive(Debug, Clone)]
+pub struct BlockRlp {
+    pub header: Option<Header>,
+}
+
+impl Decodable for BlockRlp {
+    fn decode(rlp: &rlp::Rlp) -> Result<Self, DecoderError> {
+        let header = match Header::decode_rlp(rlp) {
```
Comment (ogenev) [inline src/bin/seed-database.rs:72]: We can use here [result.ok()](https://doc.rust-lang.org/std/result/enum.Result.html#method.ok):

```rust
let header = Header::decode_rlp(rlp).ok()
```

File: src/bin/seed-database.rs [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+                                    storage.store(&content_key, &bytes_rlp).unwrap();
+                                    println!("Stored content key: {:?}", content_key_hex);
+                                    println!("- Block RLP: {:?}", prefix_removed_rlp.get(0..31));
+                                    stored_count += 1;
+                                }
+                                false => {}
+                            }
+                        }
+                    }
+                    _ => panic!("Invalid deserialized type, expected an object."),
```
Comment (ogenev) [inline src/bin/seed-database.rs:126]: What will happen if this fails in the middle of data loading? Should we think at this stage about idempotency and how we can resolve unexpected panics?

File: trin-history/src/jsonrpc.rs [hunk 4]
```diff
@@ -19,6 +22,28 @@ impl HistoryRequestHandler {
...
+                                            params.content_key
+                                        )),
+                                    },
+                                    Err(_msg) => Err(format!(
+                                        "Unable to find content key in local storage: {:?}",
+                                        params.content_key
+                                    )),
+                                }
+                            }
+                            Err(msg) => Err(format!("Invalid LocalContent params: {:?}", msg)),
```
Comment (ogenev) [inline trin-history/src/jsonrpc.rs:43]:
```suggestion
                            Err(msg) => Err(format!("Invalid LocalContent params: {msg:?}")),
```

File: trin-history/src/jsonrpc.rs [hunk 5]
```diff
@@ -19,6 +22,28 @@ impl HistoryRequestHandler {
...
+                            Ok(params) => {
+                                match &self.network.overlay.storage.get(&params.content_key) {
+                                    Ok(val) => match val {
+                                        Some(val) => Ok(Value::String(hex::encode(val.clone()))),
+                                        None => Err(format!(
+                                            "Unable to find content key in local storage: {:?}",
+                                            params.content_key
+                                        )),
+                                    },
+                                    Err(_msg) => Err(format!(
```
Comment (ogenev) [inline trin-history/src/jsonrpc.rs:37]:
```suggestion
                                    Err(_) => Err(format!(
```

File: trin-state/src/jsonrpc.rs [hunk 6]
```diff
@@ -19,6 +22,28 @@ impl StateRequestHandler {
...
+                                            params.content_key
+                                        )),
+                                    },
+                                    Err(_msg) => Err(format!(
+                                        "Unable to find content key in local storage: {:?}",
+                                        params.content_key
+                                    )),
+                                }
+                            }
+                            Err(msg) => Err(format!("Invalid LocalContent params: {:?}", msg)),
```
Comment (ogenev) [inline trin-state/src/jsonrpc.rs:43]:
```suggestion
                            Err(msg) => Err(format!("Invalid LocalContent params: {:?}"),
```

File: trin-state/src/jsonrpc.rs [hunk 7]
```diff
@@ -19,6 +22,28 @@ impl StateRequestHandler {
...
+                            Ok(params) => {
+                                match &self.network.overlay.storage.get(&params.content_key) {
+                                    Ok(val) => match val {
+                                        Some(val) => Ok(Value::String(hex::encode(val.clone()))),
+                                        None => Err(format!(
+                                            "Unable to find content key in local storage: {:?}",
+                                            params.content_key
+                                        )),
+                                    },
+                                    Err(_msg) => Err(format!(
```
Comment (ogenev) [inline trin-state/src/jsonrpc.rs:37]:
```suggestion
                                    Err(_) => Err(format!(
```

**Review by ogenev — APPROVED** [review]

File: src/bin/seed-database.rs (hunk 3 above)
Comment (njgheorghita) [reply src/bin/seed-database.rs:126]: Good point - updated this `panic!`s to `debug!`s

File: newsfragments/228.added.md [hunk 8]
```diff
@@ -0,0 +1 @@
+Add json-rpc endpoints (`portal_stateLocalContent` & `portal_historyLocalContent`) for debugging testnet nodes.
```
[the remark this answers is not in the document]
Comment (njgheorghita) [reply newsfragments/228.added.md:1]: Ummm, I'm not sure. What do you think?  This definitely feels more like a helper/debug endpoint than a function that should be accepted into the spec (i.e. all clients should eventually support).

**Pull request merged**